src/differential_privacy/mod.rs

```diff
@@ -264,27 +310,41 @@ impl Reduce {
         }
         sums = sums.group_by_iter(self.group_by().iter().cloned());
         let sums: Reduce = sums.input(self.input().clone()).build();
-        let dp_sums: Relation = sums
-            .dp_compile_sums(protected_entity_id, protected_entity_weight, epsilon, delta)?
-            .into();
-        Ok(DPRelation(output.input(dp_sums).build()))
+        let dp_sums =
+            sums.dp_compile_sums(protected_entity_id, protected_entity_weight, epsilon, delta)?;
+        Ok(DPRelation::new(
+            output.input(dp_sums.relation().clone()).build(),
+            dp_sums.private_query().clone(),
+        ))
     }
 }
 
+impl Relation {
+    fn gaussian_mechanism(self, epsilon: f64, delta: f64, bounds: Vec<(&str, f64)>) -> DPRelation {
+        let noise_multiplier = mechanisms::gaussian_noise(epsilon, delta, 1.0); // TODO set this properly
+        let noise_multipliers = bounds
```

Comment:
Why not use the Gaussian mechanism builder repeatedly?
Here everything is linear, but it may be different.